code/modules/projectiles/projectile/bullets/special.dm

```diff
@@ -38,3 +38,176 @@
+/obj/projectile/bullet/coin/proc/check_splitshot(mob/living/shooter, obj/projectile/bullet/marksman/incoming_shot, forced = FALSE)
+	if(!forced && get_dist(src, target_turf) > 1)
+		return FALSE
+
+	used = TRUE
+	var/turf/cur_tur = get_turf(src)
+	cur_tur.visible_message(span_nicegreen("[incoming_shot] impacts [src] and splits!"))
+	iterate_splitshots(shooter, incoming_shot)
+	QDEL_IN(src, 0.25 SECONDS) // may not be needed
+
+/// Now we actually create all the splitshots, loop through however many splits we'll create and fire them
+/obj/projectile/bullet/coin/proc/iterate_splitshots(mob/living/shooter, obj/projectile/incoming_shot)
+	for(var/i in 1 to num_of_splitshots)
+		fire_splitshot(incoming_shot)
+
+/// Minor convenience function for creating each shrapnel piece with circle explosions, mostly stolen from the MIRV component
+/obj/projectile/bullet/coin/proc/fire_splitshot(obj/projectile/bullet/marksman/incoming_shot)
+	var/atom/next_target = find_next_target()
+
+	ADD_TRAIT(next_target, TRAIT_RECENTLY_COINED, ABSTRACT_ITEM_TRAIT)
+	addtimer(TRAIT_CALLBACK_REMOVE(next_target, TRAIT_RECENTLY_COINED, ABSTRACT_ITEM_TRAIT), 0.5 SECONDS)
+	var/outgoing_ricoshot_level = incoming_shot.ricoshot_level + 1
+	var/obj/projectile/bullet/marksman/new_splitshot = new /obj/projectile/bullet/marksman(get_turf(src), null, outgoing_ricoshot_level)
+	//Shooting Code:
+	new_splitshot.original = next_target
+	new_splitshot.fired_from = incoming_shot.fired_from
+	new_splitshot.firer = incoming_shot.firer
+	new_splitshot.damage = 2 * incoming_shot.damage
+
+	var/current_turf = get_turf(src)
+	var/target_turf = get_turf(next_target)
+
+	if(Adjacent(current_turf, target_turf))
+		new_splitshot.fire(get_angle(current_turf, target_turf), direct_target = next_target)
+	else
+		new_splitshot.preparePixelProjectile(next_target, get_turf(src))
+		new_splitshot.fire()
+
+	if(istype(next_target, /obj/projectile/bullet/coin)) // handle further splitshot checks
+		var/obj/projectile/bullet/coin/our_coin = next_target
+		our_coin.check_splitshot(incoming_shot.firer, new_splitshot, forced = TRUE)
+
+/// Find what the splitshots will want to target next, with the order roughly based off the UK coin
+/obj/projectile/bullet/coin/proc/find_next_target()
+	var/list/valid_targets = shuffle(oview(4, loc))
+	valid_targets -= firer
+
+	for(var/obj/projectile/bullet/coin/iter_coin in valid_targets)
+		if(!iter_coin.used) // this will prevent shooting itself as well
+			return iter_coin
+
+	var/list/possible_victims = list()
+
+	for(var/mob/living/iter_living in valid_targets)
+		if(HAS_TRAIT(iter_living, TRAIT_RECENTLY_COINED) || iter_living.stat != CONSCIOUS)
+			continue
+
+		if(get_dist(src, iter_living) <= 2) // prioritize close mobs
+			return iter_living
+		possible_victims += iter_living
+
+	if(possible_victims.len)
+		return pick(possible_victims)
+
+	var/list/static/prioritized_targets = list(/obj/structure/reagent_dispensers, /obj/item/grenade, /obj/structure/window)
```

Comment:
why reagent dispensers? also does it blow up grenades?

Reply:
Yes it blows up grenades, and reagent dispensers because fuel tanks (and to a lesser degree water tanks) are funny targets that fit with ricocheting into explosives